nirum_wsgi.py

```diff
@@ -474,28 +487,123 @@ def _raw_response(self, status_code, response_json, **kwargs):
         return Response(content, status_code, headers, **kwargs)
 
 
-def compile_uri_template(template):
-    if not isinstance(template, text_type):
-        raise TypeError('template must be a Unicode string, not ' +
-                        repr(template))
-    value_pattern = re.compile('\{([a-zA-Z0-9_-]+)\}')
-    result = []
-    variables = set()
-    last_pos = 0
-    for match in value_pattern.finditer(template):
-        variable = match.group(1).replace(u'-', u'_')
-        result.append(re.escape(template[last_pos:match.start()]))
-        result.append(u'(?P<')
-        result.append(variable)
-        result.append(u'>.+?)')
-        last_pos = match.end()
-        if variable in variables:
+class UriTemplateMatchResult(object):
+
+    def __init__(self, result):
+        if result is not None:
+            self.result = List(result)
+        else:
+            self.result = None
+
+    def __bool__(self):
+        return self.result is not None
+
+    __nonzero__ = __bool__
+
+    def update(self, match_result):
+        self.result = List(
+            itertools.chain(self.result, match_result.result)
+        )
+
+    def get_variable(self, variable_name):
+        # Nirum compiler add ``_`` at the end of python keyword to avoid name
+        # duplication.  e.g. from > from_, int > int_
+        # So you have to remove ``_`` from ``variable_name``
+        # before searching for match results.
```

Comment:
Some suggestions (*italics* are edits):

> Nirum compiler *appends an underscore to* the end of *the given `variable_name` if it's a reserved keyword by Python (e.g. `from` → `from_`, `def` → `def_`).*
>
> So *we need* to remove *a trailing underscore* from the `variable_name` *(if it has one)* before *looking up* match results.

Note that `int` is not a reserved keyword but a built-in function, Nirum compiler doesn't append `_` to `int`.